packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,314 @@
+  /// or at the bottom for top to bottom layouts.
+  /// If [panePriority] is [TwoPanePriority.pane2], this is the only pane visible.
+  final Widget pane2;
+
+  /// Proportion of the screen ocupied by the first pane. The second pane takes
+  /// over the rest of the screen. A value of 0.5 will make the 2 panes equal.
+  /// This property is ignored for displays with a hinge, in which case each
+  /// pane takes over one screen.
+  final double paneProportion;
+
+  /// Same as [Flex.textDirection]
+  final TextDirection? textDirection;
+
+  /// Same as [Flex.verticalDirection]
+  ///
+  /// Defaults to [VerticalDirection.down]
+  final VerticalDirection verticalDirection;
+
+  /// Same as [Flex.direction]
+  ///
+  /// This property is ignored for displays with a hinge, in which case the
+  /// direction is [Axis.horizontal] for vertical hinges and [Axis.vertical] for
+  /// horizontal hinges.
+  ///
+  /// Defaults to [Axis.horizontal]
+  final Axis direction;
+
+  /// Whether to show only one pane and which one, or both. This is useful for
+  /// defining behaviour on narrow devices, where the 2 panes cannot be shown at
+  /// the same time.
+  ///
+  /// Defaults to [TwoPanePriority.both]
+  final TwoPanePriority panePriority;
+
+  /// The distance from the edge of the screen, used to determine how [TwoPane]
+  /// intersects [MediaQueryData.displayFeatures].
+  ///
+  /// When [TwoPane] is not the root layout and the distance from the screen
+  /// edge increases, the [padding] needs to take into account the extra space.
+  ///
+  /// For example, when TwoPane is the body of a Scaffold, the appbar adds extra
+  /// spacing between TwoPane and the edge of the screen. If TwoPane is used
+  /// with a [Axis.vertical] [direction], the separation between the two panes
+  /// will not align with [DisplayFeature]s unless the [padding] also includes
+  /// the height of the appbar.
+  ///
+  /// Defaults to [MediaQueryData.padding].
+  final EdgeInsets padding;
+
+  TextDirection? _getTextDirection(BuildContext context) =>
+      textDirection ?? Directionality.maybeOf(context);
+
+  @override
+  Widget build(BuildContext context) {
+    final MediaQueryData? mediaQuery = MediaQuery.maybeOf(context);
+    final DisplayFeature? displayFeature = _separatingDisplayFeature(mediaQuery);
+
+    late Axis _direction;
+    late int pane1Flex;
+    late int pane2Flex;
+    late Widget _pane1;
+    late Widget _pane2;
+    late Widget _delimiter;
+    final TextDirection? _textDirection = _getTextDirection(context);
+    const int fractionBase = 1000000000000;
+
+    if (mediaQuery == null || displayFeature == null) {
+      // The display is continuous, nothing is overridden.
+      _direction = direction;
+      pane1Flex = (fractionBase * paneProportion).toInt();
+      pane2Flex = fractionBase - pane1Flex;
+      _pane1 = pane1;
+      _pane2 = pane2;
+      _delimiter = Container();
```

Comment:
isn't `const SizedBox()` preferred over `Container()`? 

As `Container` can do (/does) lots of things, and when only using none or a few options it might be better (/faster) to use e.g. `SizedBox()` when using nothing or size and `DecoratedBox(decoration: ...)` when using `Container(decoration: ...)`

Reply:
You are correct. Please keep these coming, I will rebase and review this PR again on Monday 👍